packages/eslint-plugin/src/rules/no-use-before-define.ts

```diff
@@ -64,7 +64,7 @@ function isOuterEnum(
   reference: TSESLint.Scope.Reference,
 ): boolean {
   return (
-    variable.defs[0].type == DefinitionType.TSEnumName &&
+    variable.defs[0].type === DefinitionType.TSEnumName &&
```

Comment:
we really should turn on the `eqeqeq` rule

Reply:
Yeah.

1. #6227
2. #6228